feat: Provide clear error message for unknown clients

joachimvh opened the pull request:
#### 📁 Related issues

https://github.com/inrupt/solid-client-authn-js/issues/2088

#### ✍️ Description

Shows a clearer error message in case the authorizing client is unknown. This mostly happens when experimenting with new CSS instances but the client stored information in the browser that is no longer relevant. Picture below of what the output of the template looks like.

![image](https://user-images.githubusercontent.com/3447363/217566430-62829085-4b3e-4cd2-93da-e68735deac35.png)

TallTed commented on templates/error/descriptions/E0003.md.hbs:
```diff
@@ -0,0 +1,15 @@
+
+This is quite often caused by the server data being wiped,
+thereby forgetting all clients.
+After clients register they often cache this data in the browser
+and automatically try to reconnect to the server,
+but since the server no longer knows the client this request is rejected.
```
```suggestion
When clients register, they often cache this data in the browser,
and try to automatically reconnect to the server,
but since the server no longer knows the client, this request is rejected.
```

TallTed commented on templates/error/descriptions/E0003.md.hbs:
```diff
@@ -0,0 +1,15 @@
+
+The only solution to this problem is to remove the data cached by the client.
+This can be done by clearing the cookies and cache in your browser for that specific URL.
+After succes, this client wanted the server to redirect you to <{{ redirect_uri }}>,
```
```suggestion
After success, this client wanted the server to redirect you to <{{ redirect_uri }}>,
```

RubenVerborgh commented:
@joachimvh Pushed a new suggestion; untested though.
Are you sure clearing the cache is necessary? I don't see how it would help. Local storage, perhaps.

joachimvh commented:
> I don't see how it would help. Local storage, perhaps.

I meant local storage indeed. And I say both because my browser always clears both together so I'm not sure which one of the two is most relevant 😄 . Could investigate which one it is.

RubenVerborgh commented:
Yes, let's figure that out, and then this is good to go. Will already approve 👍

joachimvh commented:
Updated the steps to specify local storage needs to be removed for the client that sent them and added the redirect_url as a suggestion of what it probably is.

Also removed the option to go to the previous page as that would just cause the issue to trigger again immediately and send the user back to the error page.
